include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -415,6 +415,24 @@ class argument_parser
         return *sub_parser;
     }
 
+    /*!\brief Checks whether the option distinguished by `id` was set on the command line by the user.
+     * \param[in] id The option identifier to search for (must not contain dashes!).
+     * \returns `true` if option identifier `id` was set on the command line by the user.
+     */
+    bool option_was_set(char const id)
+    {
+        auto end = std::find(cmd_arguments.begin(), cmd_arguments.end(), "--");
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end, id) != end;
+    }
+
+    // allows all types that are implicitly convertible to std::string
+    //!\overload
+    bool option_was_set(std::string const & id)
+    {
+        auto end = std::find(cmd_arguments.begin(), cmd_arguments.end(), "--");
+        return detail::format_parse::find_option_id(cmd_arguments.begin(), end, id) != end;
+    }
```

Comment:
```suggestion
    bool option_was_set(char const id)
    {
        auto end = std::find_end(cmd_arguments.begin(), cmd_arguments.end(), "-");
        return detail::format_parse::find_option_id(cmd_arguments.begin(), end, id) != end;
    }

    // allows all types that are implicitly convertible to std::string
    //!\overload
    bool option_was_set(std::string const & id)
    {
        auto end = std::find_end(cmd_arguments.begin(), cmd_arguments.end(), "--");
        return detail::format_parse::find_option_id(cmd_arguments.begin(), end, id) != end;
    }
```

I think in your test case `{"./argument_parser_test", "-l", "hallo", "--foobar", "ballo"}` this works, because it always ends at --foobar and encloses all args.

Reply:
No this is something different.
I actually do want to search for the first occurrence of "--". Note: Not as a prefix (or other -fix) of another word but ONLY `"--"` because it signals the end of options and marks the beginning of arguments for positional options. It is special in its meaning. And since I am searching for option names I must only search before the arguments for positional options.
Was that explained in an understandable way? Otherwise I'll try again.